src/index.js

```diff
@@ -331,6 +332,22 @@ class IpfsRepo {
       })
     })
   }
+
+  /**
+   * Gets the swarm.key buffer, if it exists
+   *
+   * @param {function(Error, Buffer)} callback
+   * @returns {void}
+   */
+  swarmKey (callback) {
+    const swarmKeyKey = new Key('swarm.key')
+    this.root.get(swarmKeyKey, (err, swarmKeyBuffer) => {
```

Comment:
All keys should be on the [keychain](https://github.com/libp2p/js-libp2p-keychain)

Reply:
@richardschneider this implementation is based off of the current implementation for private network keys in go, https://github.com/ipfs/go-ipfs/blob/07feeec9c4f40f59efb57d7e16739a35b4c42d5b/repo/fsrepo/fsrepo.go#L681-L695, which are not store in the key store.

Looking at the keychain it doesn't look like it currently has the ability to store this key type. Here is the working private network spec for reference https://github.com/libp2p/specs/blob/b1365bedcd46442074fbf96610f66b0663be088a/pnet/Private-Networks-PSK-V1.md